artifactory/utils/storageinfo.go

```
@@ -0,0 +1,73 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	serviceUtils "github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+)
+
+const (
+	serviceManagerRetriesPerRequest         = 3
+	serviceManagerRetriesWaitPerRequest int = int(1 * time.Second)
+)
+
+var getRepoSummaryPollingTimeout = 10 * time.Minute
+var getRepoSummaryPollingInterval = 10 * time.Second
+
+type StorageInfoManager struct {
+	serviceManager artifactory.ArtifactoryServicesManager
+}
+
+func NewStorageInfoManager(serverDetails *config.ServerDetails) *StorageInfoManager {
+	serviceManager, err := CreateServiceManager(serverDetails, serviceManagerRetriesPerRequest, serviceManagerRetriesWaitPerRequest, false)
+	if err != nil {
+		return nil
```

We should return an error here.